[inputs.snmp] bits conversion fails

### Relevant telegraf.conf

```toml
[agent]
  snmp_translator = "gosmi"

[[inputs.snmp]]
  path = ["$MIBDIRS"]
  agents = ["10.37.155.80"]
  version = 2
  community = "public"
  agent_host_tag = "source"
  [[inputs.snmp.field]]
    oid = 'BARCO-ME-DCP-MIB::lampStatus.0'
    conversion = 'int'
    name = 'lampStatus'
```


### Logs from Telegraf

```text
% telegraf --config barco-debug2.conf --test --debug
2024-10-02T13:02:38Z I! Loading config: barco-debug2.conf
2024-10-02T13:02:38Z I! Starting Telegraf 1.32.0 brought to you by InfluxData the makers of InfluxDB
2024-10-02T13:02:38Z I! Available plugins: 235 inputs, 9 aggregators, 32 processors, 26 parsers, 62 outputs, 5 secret-stores
2024-10-02T13:02:38Z I! Loaded inputs: snmp
2024-10-02T13:02:38Z I! Loaded aggregators:
2024-10-02T13:02:38Z I! Loaded processors:
2024-10-02T13:02:38Z I! Loaded secretstores:
2024-10-02T13:02:38Z W! Outputs are not used in testing mode!
2024-10-02T13:02:38Z I! Tags enabled: host=Elliott-M2-4.local
2024-10-02T13:02:38Z D! [agent] Initializing plugins
2024-10-02T13:02:39Z W! DeprecationWarning: Value "agent_host" for option "agent_host_tag" of plugin "inputs.snmp" deprecated since version 1.29.0 and will be removed in : set to "source" for consistent usage across plugins or safely ignore this message and continue to use the current value
2024-10-02T13:02:39Z D! [agent] Starting service inputs
2024-10-02T13:02:39Z E! [inputs.snmp] Error in plugin: agent 10.37.155.80: converting "\x00" (OID .1.3.6.1.4.1.12612.220.11.2.2.4.3.0) for field lampStatus: strconv.ParseInt: parsing "\x00": invalid syntax
2024-10-02T13:02:39Z D! [agent] Stopping service inputs
2024-10-02T13:02:39Z D! [agent] Input channel closed
2024-10-02T13:02:39Z D! [agent] Stopped Successfully
2024-10-02T13:02:39Z E! [telegraf] Error running agent: input plugins recorded 1 errors
```


### System info

Telegraf 1.32, MacOS 13.6.9

### Steps to reproduce

Run this telegraf config

### Expected behavior

The device returns a bits type value of 00.  Telegraf should convert this to an integer 0.

### Actual behavior

The conversion fails.

### Additional info

This is a problem in 1.31 and 1.32 but it works fine in 1.30.3.

Below is the snmpget output from this same MacOS machine.

```
% snmpget -v2c -c public 10.37.155.80 BARCO-ME-DCP-MIB::lampStatus.0    
BARCO-ME-DCP-MIB::lampStatus.0 = BITS: 00 
```

Here's the relevant section from the MIB:
```
-- 1.3.6.1.4.1.12612.220.11.2.2.4.3
lampStatus	OBJECT-TYPE
	SYNTAX BITS
		{
			unused(0),
			on(7)
		}
	MAX-ACCESS	read-only
	STATUS	current
	DESCRIPTION
		"This is the lamp status.
		on: Lamp on"
	::= { lampProperties 3 }
```

That doesn't work either.  With `hextoint` I get this:
```
E! [inputs.snmp] Error in plugin: agent 10.37.155.80: converting "\x00" (OID .1.3.6.1.4.1.12612.220.11.2.2.4.3.0) for field lampStatus: invalid conversion type "hextoint"
```

I also tried `conversion = 'hextoint:BigEndian:uint16'` and it crashes:
```
% telegraf --test --config ~/source/mse-netboxdata/barco-debug.conf
2024-10-15T10:50:43Z I! Loading config: /Users/ebalsley/source/mse-netboxdata/barco-debug.conf
2024-10-15T10:50:43Z I! Starting Telegraf 1.32.0 brought to you by InfluxData the makers of InfluxDB
2024-10-15T10:50:43Z I! Available plugins: 235 inputs, 9 aggregators, 32 processors, 26 parsers, 62 outputs, 5 secret-stores
2024-10-15T10:50:43Z I! Loaded inputs: snmp
2024-10-15T10:50:43Z I! Loaded aggregators:
2024-10-15T10:50:43Z I! Loaded processors:
2024-10-15T10:50:43Z I! Loaded secretstores:
2024-10-15T10:50:43Z W! Outputs are not used in testing mode!
2024-10-15T10:50:43Z I! Tags enabled: host=Elliott-M2-4.local
panic: runtime error: index out of range [1] with length 1

goroutine 14 [running]:
encoding/binary.bigEndian.Uint16(...)
	encoding/binary/binary.go:146
github.com/influxdata/telegraf/internal/snmp.(*Field).Convert(0x14003bc4770, {{0x10b0b26a0, 0x14003bc37e8}, {0x140046df770, 0x23}, 0x4})
	github.com/influxdata/telegraf/internal/snmp/field.go:264 +0x1d7c
github.com/influxdata/telegraf/internal/snmp.Table.Build({{0x10900a23f, 0x4}, {0x0, 0x0, 0x0}, 0x0, {0x14002a9cee0, 0x1, 0x1}, {0x0, ...}, ...}, ...)
	github.com/influxdata/telegraf/internal/snmp/table.go:175 +0x4c8
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).gatherTable(0x140019e2780, {0x10c756200, 0x14001153820}, {0x10c713c98, 0x14003bc66e0}, {{0x10900a23f, 0x4}, {0x0, 0x0, 0x0}, ...}, ...)
	github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:134 +0xa0
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Gather.func1(0x0?, {0x14002ac02b9, 0xc})
	github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:116 +0x148
created by github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Gather in goroutine 13
	github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:102 +0x68
```

--- a/internal/snmp/field.go
+++ b/internal/snmp/field.go
@@ -253,31 +253,36 @@ func (f *Field) Convert(ent gosnmp.SnmpPDU) (interface{}, error) {
 			return v, nil
 		}
 
+		var b []byte
+		switch bit {
+		case "uint64":
+			b = make([]byte, 8)
+		case "uint32":
+			b = make([]byte, 4)
+		case "uint16":
+			b = make([]byte, 2)
+		default:
+			return nil, fmt.Errorf("invalid bit value (%s) for hex to int conversion", bit)
+		}
+		copy(b, bv)
+
+		var byteOrder binary.ByteOrder
 		switch endian {
 		case "LittleEndian":
-			switch bit {
-			case "uint64":
-				v = binary.LittleEndian.Uint64(bv)
-			case "uint32":
-				v = binary.LittleEndian.Uint32(bv)
-			case "uint16":
-				v = binary.LittleEndian.Uint16(bv)
-			default:
-				return nil, fmt.Errorf("invalid bit value (%s) for hex to int conversion", bit)
-			}
+			byteOrder = binary.LittleEndian
 		case "BigEndian":
-			switch bit {
-			case "uint64":
-				v = binary.BigEndian.Uint64(bv)
-			case "uint32":
-				v = binary.BigEndian.Uint32(bv)
-			case "uint16":
-				v = binary.BigEndian.Uint16(bv)
-			default:
-				return nil, fmt.Errorf("invalid bit value (%s) for hex to int conversion", bit)
-			}
+			byteOrder = binary.BigEndian
 		default:
 			return nil, fmt.Errorf("invalid Endian value (%s) for hex to int conversion", endian)
+		}
+
+		switch bit {
+		case "uint64":
+			v = byteOrder.Uint64(b)
+		case "uint32":
+			v = byteOrder.Uint32(b)
+		case "uint16":
+			v = byteOrder.Uint16(b)
 		}
 
 		return v, nil

--- a/internal/snmp/field_test.go
+++ b/internal/snmp/field_test.go
@@ -212,6 +212,15 @@ func TestConvertHextoint(t *testing.T) {
 			expected: uint16(0xc884),
 		},
 		{
+			name:       "little endian single byte",
+			conversion: "hextoint:LittleEndian:uint16",
+			ent: gosnmp.SnmpPDU{
+				Type:  gosnmp.OctetString,
+				Value: []byte{0x84},
+			},
+			expected: uint16(0x84),
+		},
+		{
 			name:       "little endian invalid",
 			conversion: "hextoint:LittleEndian:invalid",
 			ent:        gosnmp.SnmpPDU{Type: gosnmp.OctetString, Value: []byte{}},

